packages/@lwc/engine/src/framework/decorators/register.ts

```diff
@@ -49,7 +48,7 @@ interface WireCompilerDef {
     method?: number;
     adapter: WireAdapterConstructor;
     config: ConfigCallback;
-    params?: Record<string, any>;
+    params?: string[];
```

Comment:
Note/Ask: Params will only be a string[], since that is what the compiler returns, but... handling both `Record<string, any> | string[]` in the engine for a short period of time will greatly reduce the effort to release this change.

thoughts @pmdartus @ekashida @ravijayaramappa  ?